burn-autodiff/src/ops/tensor.rs

```diff
@@ -1147,6 +1147,32 @@ impl<B: Backend> TensorOps<ADBackendDecorator<B>> for ADBackendDecorator<B> {
         }
     }
 
+    fn abs<const D: usize>(tensor: ADTensor<B, D>) -> ADTensor<B, D> {
+        #[derive(Debug)]
+        struct Abs;
+
+        impl<B: Backend, const D: usize> Backward<B, D, 1> for Abs {
+            type State = B::TensorPrimitive<D>;
+
+            fn backward(self, ops: Ops<Self::State, 1>, grads: &mut Gradients) {
+                unary::<B, D, D, _>(ops.parents, ops.node, grads, |grad| {
+                    let input = ops.state;
+                    let abs_input = B::abs(input.clone());
+                    let value = B::div(input, abs_input);
```

Comment:
We could calculate the division directly during the forward pass and use it as the state. It would avoid recalculating the `abs` during the backward pass and doesn't increase the size of the state.

Reply:
I am new to these kinds of libraries, but wouldn't it be inconsistent?

Reply:
The API is made such that we can specilize each backward pass to have the best performance possible, so yes it is inconsistent in _implementation_ but not in _behavior_.

Reply:
To be honest I am not really sure how to do it right now. I need to dive more into the library.
I hope to have it ready in a few days

Reply:
Could you please point me to an example where this kind of optimization is done?

Reply:
Ok, now it's clear:) Thank you
I have committed suggested changes

Reply:
Do you think it would make sense to have some benchmarks run automatically, to be able to track performance changes easily?

Reply:
Yes, it makes sense to have benchmarks run automatically. We have some benches that measure ( see https://github.com/burn-rs/burn-benches) but they're run manually. One main challenge is having right machines with the devices to test different backends. Our CI uses GitHub provided machines, which is limited for full benchmarks. Perhaps, we could only test with CPU devices on provided backends.

Reply:
> Do you think it would make sense to have some benchmarks run automatically, to be able to track performance changes easily?

Filed a ticket to track it: https://github.com/burn-rs/burn/issues/573